data.sql

```diff
@@ -6,3 +6,69 @@ VALUES
   ('Gabumon', '2018-11-15', 2, true, 8),
   ('Pikachu', '2021-01-07', 1, false, 15.04),
   ('Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Charmander', '2020-02-08', 0, false, -11, 'Fire'),
+       ('Plantmon', '2021-11-15', 2, true, -5.7, 'Grass'),
+       ('Squirtle', '1993-04-02', 3, false, -12.13, 'Water'),
+       ('Angemon', '2005-06-12', 1, true, -45, 'Angel'),
+       ('Boarmon', '2005-06-07', 7, true, 20.4, 'Mammal'),
+       ('Blossom', '1998-10-13', 3, true, 17, 'Flower'),
+       ('Ditto', '2022-05-14', 4, true, 22, 'Normal');
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT new_savepoint_1;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT new_savepoint_1;
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
+/* questions */
+SELECT COUNT(*) AS total_animals
+FROM animals;
+
+SELECT COUNT(*) AS non_escape_animals
+FROM animals
+WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight
+FROM animals;
+
+SELECT neutered, COUNT(*) AS escape_count
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC
+LIMIT 1;
+
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```

Comment:
- [ ] Kindly move all these transactions and queries to the `queries.sql` file because it's supposed to be there.